pkg/oc/cli/admin/release/new.go

```diff
@@ -1466,6 +1475,32 @@ func pruneEmptyDirectories(dir string) error {
 	})
 }
 
+// ensureConsistentSources checks that images built from the same Git repository are from the same Git commit.
+func ensureConsistentSources(is *imageapi.ImageStream) error {
+	tags := make(map[string]*imageapi.TagReference, len(is.Spec.Tags))
+	for _, tag := range is.Spec.Tags {
+		location := tag.Annotations[annotationBuildSourceLocation]
+		if location == "" {
+			continue
+		}
+		previousTag, ok := tags[location]
+		if !ok {
+			tags[location] = &tag
+			continue
+		}
+		if tag.Annotations[annotationBuildSourceCommit] != previousTag.Annotations[annotationBuildSourceCommit] {
+			return fmt.Errorf(
```

Comment:
nit: it's more useful to collect all such divergence and report together, otherwise it'll be a fix one - find next game.

Reply:
> nit: it's more useful to collect all such divergence and report together, otherwise it'll be a fix one - find next game.

Hopefully this error triggers "ah, my build tooling/approach is broken, I should fix that", not one-off, per-image bumps.  But this is easy enough to implement, so done in 0a1557b5fa -> 2d094254ca.